lua/rip-substitute/popup-win.lua

```diff
@@ -310,6 +317,15 @@ function M.openSubstitutionPopup(searchPrefill)
 		group = vim.api.nvim_create_augroup("rip-substitute-popup-changes", {}),
 		callback = function()
 			ensureOnly2LinesInPopup()
+			local updatedMatches, err = rgMatches.getMatches()
+			if err then
+				vim.print(err)
+			else
+				state.matches = updatedMatches
+				--TODO: this should not always run
+				state.selectedMatch = rgMatches.getClosestMatchAfterCursor(state.matches)
+				if state.selectedMatch then rgMatches.centerViewportOnMatch(state.selectedMatch) end
+			end
```

Comment:
yeah, I think this block should not be triggered by `TextChanged`, but only on running confirming/next/prev.